doc/jinja-filters.md

```diff
@@ -0,0 +1,248 @@
+- currency: An optional currency in format 'GBP' (default), 'EUR' etc.
+
+##### Context:
+Most commonly used to display a previous answer's currency value in a following question.
+<br />
+`Of the <em>{{format_currency(answers['total-retail-turnover-answer'])}}</em> total retail turnover, what was the value of internet sales?`
+<br />
+
+
+## format_address_list
+This function format_address_list accepts two addresses, a user submitted address and a metadata address. 
+If all the items in the user entered address list are 'Undefined' (nothing in answer_store) then we retrieve the metadata.
+The address list is returned with break points between each address line.
+
+##### Parameters: 
+- user_entered_address: List of address values from answers
+- metadata_address: List of address values from metadata
+
+##### Context:
+Used to display a full address for use in a description.
+<br />
+`"description": "{{ format_address_list ([answers['address-line1'], answers['address-line2'], answers['locality'], answers['town-name'], answers['postcode']], [metadata['address_line1'], metadata['address_line2'], metadata['locality'], metadata['town_name'], metadata['postcode']]) }}"`
+<br />
+
+
+## get_current_date
+Returns the current Date.
+
+##### Parameters: 
+- N/A
+
+##### Context:
+No real current usage but a single question in Census I believe only for test purposes.
+<br />
+`The number of visitors staying in the household on {{ get_current_date() }}`
+<br />
+
+
+## format_date
+Formats a date string, returning "d MMMM YYYY", "MMMM YYYY" or "YYYY" depending on value passed. 
+
+##### Parameters: 
+- value: Value representing a datetime.
+
+##### Context:
+Used when a date is used in a question.
+<br />
+`Are you able to report for the period from {{metadata['ref_p_start_date']|format_date}} to {{metadata['ref_p_end_date']|format_date}}?`
+<br />
+
+
+## format_date_custom
+Returns a date in a defined format. In most cases in order to include the day of the week.
+
+##### Parameters: 
+- value: Value representing a datetime.
+- date_format: Format of the date to return. Default 'EEEE d MMMM YYYY'
+
+##### Context:
+Used in LMS for both Questions and Options in order to add day of week to a date. 
+Often combined with calculate_offset_from_weekday_in_last_whole_week filter.
+<br />
+`Did you have a paid job, either as an employee or self-employed, in the week {{ calculate_offset_from_weekday_in_last_whole_week(collection_metadata['started_at'], {}) | format_date_custom( 'EEEE d MMMM YYYY' ) }} to {{ calculate_offset_from_weekday_in_last_whole_week(collection_metadata['started_at'], {}, 'SU') | format_date_custom( 'EEEE d MMMM YYYY' ) }}`
+ 
+`"label": "{{calculate_offset_from_weekday_in_last_whole_week(collection_metadata['started_at'], {}, 'MO') | format_date_custom('EEEE dd MMMM') }}"`
+<br />
+
+
+## format_conditional_date
+This is to be used when passing multiple dates as multiple arguments. It will return the first valid date formatted.
```

Comment:
Possibly not for this pull request, but I wonder now that we have `first_non_empty_item` whether we could combine it with `format_date` so we don't need this filter, something like:
```
{{ first_non_empty_item(answers['period-from'], metadata['ref_p_start_date']) | format_date }}
```

Reply:
I agree and thought exactly the same. I'd like to check with Mebin again as there was a reason he didn't do it at the time of writing this new filter.

Reply:
So `format_conditional_date` reuses the same private method that `first_non_empty_item` uses `_get_first_non_empty_item`.

Having it as two filter just mean't I didn't have to essentially duplicate
``` python
	date_format = 'd MMMM YYYY'
    if value and re.match(r'\d{4}-\d{2}$', value):
        date_format = 'MMMM YYYY'
    if value and re.match(r'\d{4}$', value):
        date_format = 'YYYY'
```
to check whether what it got passed in regex matched any one of the different formats we support before passing it into the `format_date` filter and then just returning the value if it wasn't a date. I guess you could also try: except to see if it would parse as a date. 🤷‍♂️ 

Just wasn't sure if this would have be very nice.